app/src/main/java/pl/droidevs/books/repository/BookRepository.java

```diff
@@ -0,0 +1,32 @@
+package pl.droidevs.books.repository;
+
+import android.arch.lifecycle.LiveData;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import pl.droidevs.books.app.BookDataBase;
+import pl.droidevs.books.entity.Book;
+
+public class BookRepository {
+
+    @Inject
+    BookDataBase bookDataBase;
+
+    public BookRepository(BookDataBase bookDataBase) {
+        this.bookDataBase = bookDataBase;
+    }
+
+    public void addBook(Book book) {
+        bookDataBase.bookDao().addBook(book);
+    }
+
+    public void removeBook(Book book) {
+        bookDataBase.bookDao().removeBook(book);
+    }
+
+    public LiveData<List<Book>> getBooks() {
```

Comment:
Czy repozytorium powinno zwracać encję bazodanową? Moim zdaniem nie, ponieważ: 1. repozytorium ma synchronizować różne źródła danych, więc dane z bazy mogą różnić się od danych z webserwisu 2. nie wszystkie pola w bazie danych mogą być nam potrzebne w naszje domenie. Możemy sobie wyobrazić, że w bazie danych będzie pole: creationDate po którym sortowane będą wyniki, a nasza domena nie musi nic wiedzieć o jego istnieniu 3. obiekt domenowy może być typowym obiektem z OOP, a nie tylko strukturą danych.

Reply:
Ok, to oznacza że potrzebny jest jakiś mapper z jednego na drugi. Stworzyłam coś tam, ale czy nie można lepiej? - not sure.